From 7d2ba3617da211ad9fe3cdcf036a0500e4765c6e Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Sun, 30 Jun 2019 22:48:42 +0800
Subject: [PATCH 2/2] arm64: allwinner: a64: dts: add volumn keys support for
 PineTab

PineTab has a pair of volumn keys (up and down) which are connected via
a resistor ladder on the LRADC pin of the A64 SoC. Volumn + key is
active at 0.2V and Volumn - is at 0.4V according to calculation.

Add volumn keys support for PineTab by enabling the LRADC and fill key
voltage infomation.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 .../boot/dts/allwinner/sun50i-a64-pinetab.dts | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinetab.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinetab.dts
index 5a714429ec08..fe4a2e58f2d9 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinetab.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinetab.dts
@@ -194,6 +194,25 @@
 	bias-pull-up;
 };
 
+&lradc {
+	vref-supply = <&reg_aldo3>;
+	status = "okay";
+
+	button-200 {
+		label = "Volume Up";
+		linux,code = <KEY_VOLUMEUP>;
+		channel = <0>;
+		voltage = <200000>;
+	};
+
+	button-400 {
+		label = "Volume Down";
+		linux,code = <KEY_VOLUMEDOWN>;
+		channel = <0>;
+		voltage = <400000>;
+	};
+};
+
 &mixer1 {
 	status = "okay";
 };
-- 
2.21.0

